Mathlib/Algebra/Polynomial/Roots.lean

```diff
@@ -785,6 +822,26 @@ theorem map_roots_le_of_injective [IsDomain A] [IsDomain B] (p : A[X]) {f : A 
   · simp only [hp0, roots_zero, Multiset.map_zero, Polynomial.map_zero, le_rfl]
   exact map_roots_le ((Polynomial.map_ne_zero_iff hf).mpr hp0)
 
+theorem map_roots_card_le_degree {A B : Type*} [Semiring A] [CommRing B] [IsDomain B]
+    {f : A →+* B} (p : A[X]) (h : p.map f ≠ 0) : (p.map f).roots.card ≤ p.degree :=
+  card_roots h |>.trans degree_map_le
+
+theorem map_roots_card_le_natDegree {A B : Type*} [Semiring A] [CommRing B] [IsDomain B]
+    {f : A →+* B} (p : A[X]) : (p.map f).roots.card ≤ p.natDegree :=
+  card_roots' _ |>.trans natDegree_map_le
+
+open Classical in
+theorem map_roots_filter_range_eq_roots_map [IsDomain A] [IsDomain B]
+    {f : A →+* B} (hf : Function.Injective f) (p : A[X]) :
+    (p.map f).roots.filter (· ∈ f.range) = p.roots.map f := by
```

Comment:
Please assume the `DecidablePred` instance instead of `open`ing `Classical`.

Reply:
This is to avoid "diamond", i.e. right now if someone provides another instance that the predicate is decidable, then they will not be able to use your theorem.